### PR TITLE
Add BoundReferences to ComputedTrafficPermissions

### PR DESCRIPTION
**Pull request by kisunji**
### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: internal/auth/internal/controllers/trafficpermissions/controller_test.go [hunk 1]
```diff
@@ -729,6 +732,91 @@ func (suite *controllerSuite) TestReconcile_TrafficPermissionsDelete_Destination
 	})
 }
 
+// 1. Create ALLOW traffic permission granting foo -> bar
+// 2. Observe reconciler write CTP for bar listing source foo
+// 3. User updates TP from step 1 to instead grand foo -> baz
+// 4. Observe reconciler update CTP for bar to list source baz
+// 5. (must) Observe reconciler update CTP for bar to default (no permissions)
+func TestController_OrphanedTrafficPermissions(t *testing.T) {
```
Comment (kisunji) [inline internal/auth/internal/controllers/trafficpermissions/controller_test.go:740]: This is a standalone "controller test" that `TestController*` tests below should be updated to follow.
It does not require manual `Reconcile` calls and tests the lifecycle of the controller in response to events (like resource writes in this test).

Comment (kisunji) [reply internal/auth/internal/controllers/trafficpermissions/controller_test.go:740]: On another note, this test will pass even without the BRC code added as part of this PR. The bimapper used by the controller [here](https://github.com/hashicorp/consul/blob/202090e5d55c93f9170890bc2303e5853d147c3c/internal/auth/internal/mappers/trafficpermissionsmapper/traffic_permissions_mapper.go#L52-L55) already handles orphaned CTPs for us.
However, we plan to phase out bimappers entirely so this test is a sanity check that the cleanups still happen. 

I have confirmed that if I comment out the loop to handle old CTPs linked above, this test will fail without the BRC code.

File: internal/auth/internal/types/computed_traffic_permissions.go [hunk 2]
```diff
@@ -67,5 +67,6 @@ func aclReadHookComputedTrafficPermissions(authorizer acl.Authorizer, authzConte
 }
 
 func aclWriteHookComputedTrafficPermissions(authorizer acl.Authorizer, authzContext *acl.AuthorizerContext, res *pbresource.Resource) error {
-	return authorizer.ToAllowAuthorizer().TrafficPermissionsWriteAllowed(res.Id.Name, authzContext)
+	// users should not be writing computed resources
+	return authorizer.ToAllowAuthorizer().OperatorWriteAllowed(authzContext)
```
Comment (kisunji) [inline internal/auth/internal/types/computed_traffic_permissions.go:71]: Test failing on ACL changes. Will fix

Comment (kisunji) [reply internal/auth/internal/types/computed_traffic_permissions.go]: There is another PR refactoring ACL tests that hasn't been merged yet. Will remove ACLs from this PR until that other PR gets merged

**Review by rboyer — APPROVED** [review]

**Pull request merged**